tests/unit/dogstatsd/test_statsd.py

```diff
@@ -328,6 +336,26 @@ def test_counter(self):
         self.statsd.flush()
         self.assert_equal_telemetry('page.views:-12|c\n', self.recv(2))
 
+    def test_count(self):
+        self.statsd.count('page.views', 11)
+        self.statsd.flush()
+        self.assert_equal_telemetry('page.views:11|c\n', self.recv(2))
+
+    def test_count_with_ts(self):
+        self.statsd.count_with_timestamp("page.views", 1, timestamp=1066)
+        self.statsd.flush()
+        self.assert_equal_telemetry("page.views:1|c|T1066\n", self.recv(2))
+
+        self.statsd._reset_telemetry()
+        self.statsd.count_with_timestamp("page.views", 11, timestamp=2121)
+        self.statsd.flush()
+        self.assert_equal_telemetry("page.views:11|c|T2121\n", self.recv(2))
+
+    def test_count_with_invalid_ts_should_be_ignored(self):
+        self.statsd.count_with_timestamp("page.views", 1, timestamp=-1066)
+        self.statsd.flush()
+        self.assert_equal_telemetry("page.views:1|c\n", self.recv(2))
+
```

Comment:
Would that be worth a direct call to `_report` with a timestamp and a wrong metric type to make sure it sets the timestamp to `0`? Not sure.

Reply:
Yeah, I feel there should be one more test to make sure that if statement that checks metric types with timestamps is covered correctly.